app/screens/sso/index.tsx

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {injectIntl, intlShape} from 'react-intl';
+import {shallowEqual, useDispatch, useSelector} from 'react-redux';
+
+import type {GlobalState} from '@mm-redux/types/store';
+
+import {resetToChannel} from 'app/actions/navigation';
+import {ViewTypes} from 'app/constants';
+import tracker from 'app/utils/time_tracker';
+import {scheduleExpiredNotification} from '@actions/views/session';
+import {ssoLogin} from '@actions/views/user';
+import {DispatchFunc} from '@mm-redux/types/actions';
+import {Client4} from '@mm-redux/client';
+import {getTheme} from '@mm-redux/selectors/entities/preferences';
+import {getConfig} from '@mm-redux/selectors/entities/general';
+import {ErrorApi} from '@mm-redux/types/client4';
+import {isMinimumServerVersion} from '@mm-redux/utils/helpers';
+
+import SSOWithRedirectURL from './sso_with_redirect_url';
+import SSOWithWebView from './sso_with_webview';
+
+interface SSOProps {
+    intl: typeof intlShape;
+    ssoType: string;
+}
+
+function SSO({intl, ssoType}: SSOProps) {
+    const [config, serverUrl, theme] = useSelector((state: GlobalState) => ([
+        getConfig(state),
+        state.views.selectServer.serverUrl,
+        getTheme(state),
+    ]), shallowEqual);
+
+    const [loginError, setLoginError] = React.useState<string>('');
+
+    const asyncDispatch: DispatchFunc = useDispatch();
+    const dispatch = useDispatch();
+
+    let completeUrlPath = '';
+    let loginUrl = '';
+    switch (ssoType) {
+    case ViewTypes.GOOGLE: {
+        completeUrlPath = '/signup/google/complete';
+        loginUrl = `${serverUrl}/oauth/google/mobile_login`;
+        break;
+    }
+    case ViewTypes.GITLAB: {
+        completeUrlPath = '/signup/gitlab/complete';
+        loginUrl = `${serverUrl}/oauth/gitlab/mobile_login`;
+        break;
+    }
+    case ViewTypes.SAML: {
+        completeUrlPath = '/login/sso/saml';
+        loginUrl = `${serverUrl}/login/sso/saml?action=mobile`;
+        break;
+    }
+    case ViewTypes.OFFICE365: {
+        completeUrlPath = '/signup/office365/complete';
+        loginUrl = `${serverUrl}/oauth/office365/mobile_login`;
```

Comment:
This endpoint is new right? If it is, please handle accordingly

Reply:
Yes it is a new one and it supports both new and old implementation.

Reply:
Sorry this is not resolved, if I use this version of the mobile app against an old server, it will show a 404 as this endpoint does not exist in old servers, or I'm I missing something?

Reply:
https://github.com/mattermost/mattermost-server/blob/caeda90b246b2830ffc29901efce685a08d638fa/web/oauth.go#L31

`${serverUrl}/oauth/{gitlab/google/office365}/mobile_login`.

It's a regex on server side which will handle office365 as well. We have tested it on 5.25 as well and it's working.

Reply:
> Yes it is a new one and it supports both new and old implementation.

Sorry I meant it's a "NEW ONE" added to the mobile but it was already supported by the server.

Reply:
Oh ok, perfect then!! Thanks for clarifying ;)